api-docs/guide/adding-a-custom-udf.rst

```diff
@@ -0,0 +1,88 @@
+.. _guide-adding-a-custom-udf:
+
+Adding a custom UDF 
+===================
+
+    This section is an overview of how you can create and execute a custom UDF. 
+
+Introduction to UDFs
+=====================
+
+    * UDFs are user-defined functions, used to achieve a specific functionality on any video loaded to EVA. 
+    * As the name suggests, these functions are defined and implemented by users. Once the UDFs are registered, you can 'call' these functions in your queries. 
+
+Implementing your own UDF
+=========================
+
+    1. Create a new file under `src/udfs/` and give it a descriptive name. eg: `fastrcnn_object_detector.py`, `midas_depth_estimator.py`. 
+    
+    2. Here, you need to create and implement a class that derives from `PytorchAbstractUDF`.  
+
+        * The `PytorchAbstractUDF` is a parent class that defines and implements standard methods for model inference.
+
+        * `_get_predictions()` - an abstract method that needs to be implemented in your child class.
+
+        * `classify()` - A  method that receives the frames and calls the `_get_predictions()` implemented in your child class. Based on GPU batch size, it also decides whether to split frames into chunks and performs the accumulation.
+
```

Comment:
We should also provide a few example code snippets so that readers can follow. Add examples from already existing one.

Reply:
Got it. I'll do that.